File: insertion_sort.swift
#!/usr/bin/env swift
// http://rosettacode.org/wiki/Sorting_algorithms/Insertion_sort

func insertionSort(s: [Int]) -> [Int] {
  var sortable = s
  for i in 1..<s.count {
    let val = sortable[i]
    var j = i - 1
    while j >= 0 && sortable[j] > val {
      sortable[j+1] = sortable[j]
      j--
    }
    sortable[j+1] = val
  }
  return sortable
}

let arrayToSort = [100, 2, 56, 200, -52, 3, 99, 33, 177, -199]
print(insertionSort(arrayToSort))

File: insertion_sort.py
#!/usr/bin/env python

def insertion_sort(list):
  new_list = list
  for i in xrange(0,len(new_list)):
    v = new_list[i]
    j = i-1
    while j >= 0 and new_list[j] > v:
      new_list[j+1] = new_list[j]
      j = j - 1
    new_list[j+1] = v
  return new_list

t = [4, 65, 2, -31, 0, 99, 2, 83, 782, 1]

print insertion_sort(t)